packages/functional-tests/tests/react-conversion/signup.spec.ts

```diff
@@ -0,0 +1,154 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import { expect, newPagesForSync, test } from '../../lib/fixtures/standard';
+import { EmailHeader, EmailType } from '../../lib/email';
+import { createCustomEventDetail, FirefoxCommand } from '../../lib/channels';
+
+const PASSWORD = 'passwordzxcv';
+
+test.describe('signup react', () => {
+  let email;
+  test.beforeEach(async ({ pages: { login, configPage } }) => {
+    test.slow();
+    // Ensure that the feature flag is enabled
+    const config = await configPage.getConfig();
+    test.skip(config.showReactApp.signUpRoutes !== true);
+
+    email = login.createEmail('signup_react{id}');
+  });
+
+  test.afterEach(async ({ target }) => {
+    if (email) {
+      // Cleanup any accounts created during the test
+      try {
+        await target.auth.accountDestroy(email, PASSWORD);
+      } catch (e) {
+        // Ignore errors
+      }
+    }
+  });
+
+  test('signup web', async ({ page, target, pages: { signupReact } }) => {
+    await signupReact.goto();
+    await signupReact.fillOutEmailFirst(email);
+    await signupReact.fillOutSignupForm(PASSWORD);
+
+    const code = await target.email.waitForEmail(
+      email,
+      EmailType.verifyShortCode,
+      EmailHeader.shortCode
+    );
+
+    await signupReact.fillOutCodeForm(code);
+
+    // Verify logged into settings page
+    await page.waitForURL(/settings/);
+  });
+
+  test('signup oauth', async ({
```

Comment:
The oauth, webchannel and sync tests have the main bits in place to test, however we are still working on those flows so wil have to tweak this.